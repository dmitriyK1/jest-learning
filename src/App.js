import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Message from './Message';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  static propTypes = {
    value: PropTypes.number,
  };

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <input type="text" value={this.props.value} />
          <Message />
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}

export default App;
