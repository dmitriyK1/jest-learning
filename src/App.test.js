import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import jest from 'jest';
import render from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);

  const temp = render.create(<App />).toJSON();
    console.log(temp);

  expect(temp).toMatchSnapshot();
});
