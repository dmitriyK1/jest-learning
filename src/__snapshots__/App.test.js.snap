// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component tests snapshot matches to stored snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="test-file-stub"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;
